crates/biome_service/src/file_handlers/mod.rs

```diff
@@ -359,40 +358,46 @@ impl DocumentFileSource {
     }
 }
 
-impl biome_console::fmt::Display for DocumentFileSource {
-    fn fmt(&self, fmt: &mut Formatter) -> std::io::Result<()> {
+impl std::fmt::Display for DocumentFileSource {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter) -> std::fmt::Result {
         match self {
             Self::Js(js) => {
                 let is_jsx = js.is_jsx();
                 if js.is_typescript() {
                     if is_jsx {
-                        fmt.write_markup(markup! { "TSX" })
+                        write!(fmt, "TSX")
                     } else {
-                        fmt.write_markup(markup! { "TypeScript" })
+                        write!(fmt, "TypeScript")
                     }
                 } else if is_jsx {
-                    fmt.write_markup(markup! { "JSX" })
+                    write!(fmt, "JSX")
                 } else {
-                    fmt.write_markup(markup! { "JavaScript" })
+                    write!(fmt, "JavaScript")
                 }
             }
             Self::Json(json) => {
                 if json.allow_comments() {
-                    fmt.write_markup(markup! { "JSONC" })
+                    write!(fmt, "JSONC")
                 } else {
-                    fmt.write_markup(markup! { "JSON" })
+                    write!(fmt, "JSON")
                 }
             }
-            Self::Css(_) => fmt.write_markup(markup! { "CSS" }),
-            Self::Graphql(_) => fmt.write_markup(markup! { "GraphQL" }),
-            Self::Html(_) => fmt.write_markup(markup! { "HTML" }),
-            Self::Grit(_) => fmt.write_markup(markup! { "Grit" }),
-            Self::Ignore => fmt.write_markup(markup! { "Ignore" }),
-            Self::Unknown => fmt.write_markup(markup! { "Unknown" }),
+            Self::Css(_) => write!(fmt, "CSS"),
+            Self::Graphql(_) => write!(fmt, "GraphQL"),
+            Self::Html(_) => write!(fmt, "HTML"),
+            Self::Grit(_) => write!(fmt, "Grit"),
+            Self::Ignore => write!(fmt, "Ignore"),
+            Self::Unknown => write!(fmt, "Unknown"),
         }
     }
 }
 
+impl biome_console::fmt::Display for DocumentFileSource {
+    fn fmt(&self, fmt: &mut Formatter) -> std::io::Result<()> {
+        fmt.write_str(self.to_string().as_str())
```

Comment:
Any chance to avoid the allocation of a string?

Reply:
I'd like to, but I don't know how to deal with the biome_console formatter. Is there something like an adapter between `std::fmt::Formatter` and `biome_console::fmt::Formatter`?

Reply:
There's also the type `MessageAndDescription`, which takes care of it